OsmApi.pm

```diff
@@ -143,6 +143,7 @@ sub put
     return dummylog("PUT", $url, $body) if ($prefs->{dryrun});
     my $req = HTTP::Request->new(PUT => $prefs->{apiurl}.$url);
     $req->header("Content-type" => "text/xml");
+    $req->header("Authorization" => "Basic ".encode_base64($prefs->{username}.":".$prefs->{password}));
```

Comment:
@woodpeck Honestly I'm not sure. When I ran the script in the production osm api it didn't seem to be setting the authorisation and this fixed it, but again I didn't look too deeply into if this is change correct or even really necessary, it was rather a quick fix.

I think the previous commit for handling passwords with spaces and/or equals chars should get merged if you're okay with the change. But I'm not sure about this commit.